etc/tests/test_bootstrapper.py

```diff
@@ -0,0 +1,289 @@
+        with pytest.raises(minio.error.InvalidEndpointError):
+            bootstrapper.main()
+
+
+def test_fail_bad_notebook_main_method(monkeypatch, s3_setup, tmpdir):
+    argument_dict = {'cos-endpoint': 'http://127.0.0.1:9000',
+                     'cos-bucket': 'test-bucket',
+                     'cos-directory': 'test-directory',
+                     'cos-dependencies-archive': 'test-bad-archiveB.tgz',
+                     'notebook': 'etc/tests/resources/test-bad-notebookB.ipynb',
+                     'inputs': 'test-file.txt',
+                     'outputs': 'test-file.txt'}
+    bucket_name = "test-bucket"
+
+    monkeypatch.setattr(bootstrapper, "parse_arguments", lambda x: argument_dict)
+    monkeypatch.setattr(bootstrapper, "package_install", lambda: True)
+
+    mocked_func = mock.Mock(return_value="default", side_effect=['test-bad-archiveB.tgz',
+                                                                 'test-file.txt',
+                                                                 'test-bad-notebookB-output.ipynb',
+                                                                 'test-bad-notebookB.html',
+                                                                 'test-file.txt'])
+    monkeypatch.setattr(bootstrapper, "get_object_storage_filename", mocked_func)
+
+    monkeypatch.setenv("AWS_ACCESS_KEY_ID", "minioadmin")
+    monkeypatch.setenv("AWS_SECRET_ACCESS_KEY", "minioadmin")
+
+    s3_setup.fput_object(bucket_name=bucket_name,
+                         object_name="test-file.txt",
+                         file_path="README.md")
+    s3_setup.fput_object(bucket_name=bucket_name,
+                         object_name="test-bad-archiveB.tgz",
+                         file_path="etc/tests/resources/test-bad-archiveB.tgz")
+
+    with tmpdir.as_cwd():
+        with pytest.raises(papermill.exceptions.PapermillExecutionError):
+            bootstrapper.main()
+
+
+def test_package_installation(monkeypatch):
+    elyra_dict = {'ipykernel': '5.3.0',
+                  'ansiwrap': '0.8.5',
+                  'package': '1.0.0'}
+    current_dict = {'bleach': '3.1.5',
+                    'ansiwrap': '0.8.4',
+                    'package': '1.0.2',
+                    'text-extensions-for-pandas':
+                        "git+https://github.com/test/text-extensions-for-pandas@a0dcb9196c6de6a2f58194cc49e48a25a18d099d"}
+    mocked_func = mock.Mock(return_value="default", side_effect=[elyra_dict, current_dict])
+
+    monkeypatch.setattr(bootstrapper, "package_list_to_dict", mocked_func)
+    monkeypatch.setattr(subprocess, "check_call", lambda x: True)
+
+    bootstrapper.package_install()
+
+
+def test_convert_notebook_to_html(tmpdir):
+    notebook_file = os.getcwd() + "/etc/tests/resources/test-notebookA.ipynb"
+    notebook_output_html_file = "notebookA.html"
+    with tmpdir.as_cwd():
+        bootstrapper.convert_notebook_to_html(notebook_file, notebook_output_html_file)
+
```

Comment:
Add confirmation.  Does the file exist?  Does certain expected content exist in the output?

Reply:
going to check for file exist and hash compare for content